cmd/kubeseal/main.go

```diff
@@ -700,28 +752,28 @@ func run(clientConfig clientcmd.ClientConfig, ctx context.Context, w io.Writer,
 		w = f
 	}
 
-	if unseal {
-		return unsealSealedSecret(w, input, scheme.Codecs, privKeys)
+	if flags.unseal {
+		return unsealSealedSecret(flags, w, input, scheme.Codecs, flags.privKeys)
 	}
-	if len(privKeys) != 0 && isatty.IsTerminal(os.Stderr.Fd()) {
+	if len(flags.privKeys) != 0 && isatty.IsTerminal(os.Stderr.Fd()) {
 		fmt.Fprintf(os.Stderr, "warning: ignoring --recovery-private-key because unseal command not chosen with --recovery-unseal\n")
 	}
 
-	if validateSecret {
-		return validateSealedSecret(clientConfig, ctx, input, controllerNs, controllerName)
+	if flags.validateSecret {
+		return validateSealedSecret(cfg, input, flags.controllerNs, flags.controllerName)
```

Comment:
Looking the code, we are passing the flags as parameters. Maybe one future improvement is to use the flag struct as variable and then use the values inside each function. We can reduce the parameters in the functions. WDYT?

Reply:
Need to review this carefully.

In some places the call used the flag values as fallbacks and the explicit arguments as the overrides. Not that I like that, but it seems something for a follow up PR.

In this particular case, it seems we could remove those arguments and just pass the config (contains the flags) and the input. As I said  think that is better for a follow up PR. Specially as if we move this code around into a library we might not want to pass cfg or flags directly.

Reply:
I guess we can already fix this particular case here and worry about further refactors later.

Will fix.

Reply:
Done, I also fixed `reEncryptSealedSecret` which was NOT using overrides as well.